src/Compilers/CSharp/Portable/Compiler/Instrumentation.cs

```diff
@@ -157,13 +183,37 @@ public override BoundNode Visit(BoundNode node)
                     case BoundKind.TryStatement:
                     // A for statement implicitly gets coverage for the initial statements and the increment statement.
                     case BoundKind.ForStatement:
+                        return visited;
+                    case BoundKind.ReturnStatement:
+                        if (!_methodHasExplicitBlock && ((BoundReturnStatement)statement).ExpressionOpt != null)
+                        {
+                            // The return statement for value-returning methods defined without a block is compiler generated, but requires instrumentation.
+                            return ForceCollectDynamicAnalysis(visited);
+                        }
+                        break;
+                    case BoundKind.ExpressionStatement:
+                        if (!_methodHasExplicitBlock)
+                        {
+                            // The assignment statement for a property set method defined without a block is compiler generated, but requires instrumentation.
+                            return ForceCollectDynamicAnalysis(visited);
+                        }
+                        break;
+                    case BoundKind.LocalDeclaration:
+                        if (statement.Syntax.Parent.Kind() == SyntaxKind.VariableDeclaration && ((VariableDeclarationSyntax)statement.Syntax.Parent).Variables.Count > 1)
```

Comment:
It can happen during speculative binding. If that code path doesn't run through here, then the null check isn't required.